### PR TITLE
feature/add check department logic

### PR DESCRIPTION
**Pull request by borovHokage**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/837

### CODE REVIEW

File: src/CompanyService.Broker/Consumers/CheckDepartmentsExistence.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            await context.RespondAsync<IOperationResult<ICheckDepartmentsExistence>>(response);
+        }
+
+        private object GetDepartmentExistenceInfo(ICheckDepartmentsExistence requestIds)
+        {
+            List<Guid> departmentIds = _repository.AreDepartmentsExist(requestIds.DepartmentIds);
+
+            return ICheckDepartmentsExistence.CreateObj(departmentIds);
+        }
+
```
Comment (MariyaSin) [inline src/CompanyService.Broker/Consumers/CheckDepartmentsExistence.cs:33]: remove line

File: src/CompanyService.Broker/Consumers/CheckDepartmentsExistence.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
```
Comment (MariyaSin) [inline src/CompanyService.Broker/Consumers/CheckDepartmentsExistence.cs:35]: remove line

File: src/CompanyService.Broker/Consumers/CheckDepartmentsExistence.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            var response = OperationResultWrapper.CreateResponse(GetDepartmentExistenceInfo, context.Message);
+
+            await context.RespondAsync<IOperationResult<ICheckDepartmentsExistence>>(response);
+        }
+
+        private object GetDepartmentExistenceInfo(ICheckDepartmentsExistence requestIds)
+        {
+            List<Guid> departmentIds = _repository.AreDepartmentsExist(requestIds.DepartmentIds);
+
+            return ICheckDepartmentsExistence.CreateObj(departmentIds);
```
Comment (BelehovEgor) [inline src/CompanyService.Broker/Consumers/CheckDepartmentsExistence.cs:31]: you can do it without variable

File: src/CompanyService.Data/DepartmentRepository.cs [hunk 4]
```diff
@@ -70,6 +70,13 @@ public DbDepartment GetDepartment(Guid? departmentId, Guid? userId)
             return dbDepartment;
         }
 
+        public List<Guid> AreDepartmentsExist(List<Guid> departmentId)
+        {
+            return _provider.Departments
+                .Where(d => departmentId.Contains(d.Id) && d.IsActive)
+                .Select(d => d.Id).ToList();
```
Comment (BelehovEgor) [inline src/CompanyService.Data/DepartmentRepository.cs:77]: mb you should move ToList to next line?

**Review by BelehovEgor — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

File: src/CompanyService.Broker/Consumers/CheckDepartmentsExistence.cs (hunk 2 above)
Comment (borovHokage) [reply src/CompanyService.Broker/Consumers/CheckDepartmentsExistence.cs]: done

File: src/CompanyService.Broker/Consumers/CheckDepartmentsExistence.cs (hunk 3 above)
Comment (borovHokage) [reply src/CompanyService.Broker/Consumers/CheckDepartmentsExistence.cs]: fixed

File: src/CompanyService.Data/DepartmentRepository.cs (hunk 4 above)
Comment (borovHokage) [reply src/CompanyService.Data/DepartmentRepository.cs]: done

File: src/CompanyService.Broker/Consumers/CheckDepartmentsExistence.cs (hunk 1 above)
Comment (borovHokage) [reply src/CompanyService.Broker/Consumers/CheckDepartmentsExistence.cs]: done

File: src/CompanyService.Data/Interfaces/IDepartmentRepository.cs [hunk 5]
```diff
@@ -28,6 +28,8 @@ public interface IDepartmentRepository
 
         DbDepartment GetDepartment(GetDepartmentFilter filter);
 
+        List<Guid> AreDepartmentsExist(List<Guid> departmentId);
```
Comment (MariyaSin) [inline src/CompanyService.Data/Interfaces/IDepartmentRepository.cs:31]: for list write departmentIds

File: src/CompanyService.Data/DepartmentRepository.cs [hunk 6]
```diff
@@ -70,6 +70,14 @@ public DbDepartment GetDepartment(Guid? departmentId, Guid? userId)
             return dbDepartment;
         }
 
+        public List<Guid> AreDepartmentsExist(List<Guid> departmentId)
```
Comment (MariyaSin) [inline src/CompanyService.Data/DepartmentRepository.cs:73]: for list write departmentIds

**Review by MariyaSin — APPROVED** [review]

File: src/CompanyService.Data/Interfaces/IDepartmentRepository.cs (hunk 5 above)
Comment (borovHokage) [reply src/CompanyService.Data/Interfaces/IDepartmentRepository.cs]: Fixed

**Review by djino97 — APPROVED** [review]

**Pull request merged**